functions/auth0.py

```diff
@@ -88,12 +89,36 @@ def _get_log_bucket():
 
 def _get_last_auth0_log_id() -> Optional[str]:
     """Fetch that ID of the last access log imported from Auth0"""
-    blob = _get_log_bucket().get_blob(LAST_LOG_ID)
-    if blob:
-        return blob.download_as_string().decode("utf-8")
+    bucket = _get_log_bucket()
+    blobs = bucket.list_blobs(prefix=LAST_LOG_ID_DIR)
+
+    last_blob = None
+
+    for blob in blobs:
+        if not last_blob:
+            last_blob = blob
+            continue
+
+        if blob.name > last_blob.name:
+            last_blob = blob
```

Comment:
```suggestion
    # taking alphabetically last blob, because they are named with iso formatted dates, 
    # (see _get_new_last_auth0_log_id_blob_name) thus we need the last one
    last_blob = max(blobs, key=lambda b: b.name) if blobs else None
```